controller_manager/test/test_controllers_chaining_with_controller_manager.cpp

```diff
@@ -610,6 +679,271 @@ TEST_P(
   ASSERT_FALSE(diff_drive_controller->is_in_chained_mode());
 }
 
+TEST_P(
+  TestControllerChainingWithControllerManager, test_chained_controllers_activation_error_handling)
+{
+  SetupControllers();
+
+  // add all controllers - CONTROLLERS HAVE TO ADDED IN EXECUTION ORDER
+  cm_->add_controller(
+    position_tracking_controller, POSITION_TRACKING_CONTROLLER,
+    test_chainable_controller::TEST_CONTROLLER_CLASS_NAME);
+  cm_->add_controller(
+    diff_drive_controller, DIFF_DRIVE_CONTROLLER,
+    test_chainable_controller::TEST_CONTROLLER_CLASS_NAME);
+  cm_->add_controller(
+    diff_drive_controller_two, DIFF_DRIVE_CONTROLLER_TWO,
+    test_chainable_controller::TEST_CONTROLLER_CLASS_NAME);
+  cm_->add_controller(
+    pid_left_wheel_controller, PID_LEFT_WHEEL,
+    test_chainable_controller::TEST_CONTROLLER_CLASS_NAME);
+  cm_->add_controller(
+    pid_right_wheel_controller, PID_RIGHT_WHEEL,
+    test_chainable_controller::TEST_CONTROLLER_CLASS_NAME);
+
+  CheckIfControllersAreAddedCorrectly();
+
+  ConfigureAndCheckControllers();
+
+  // Set ControllerManager into Debug-Mode output to have detailed output on updating controllers
+  cm_->get_logger().set_level(rclcpp::Logger::Level::Debug);
+  rclcpp::get_logger("ControllerManager::utils").set_level(rclcpp::Logger::Level::Debug);
+
+  // at beginning controllers are not in chained mode
+  EXPECT_FALSE(pid_left_wheel_controller->is_in_chained_mode());
+  EXPECT_FALSE(pid_right_wheel_controller->is_in_chained_mode());
+  ASSERT_FALSE(diff_drive_controller->is_in_chained_mode());
+
+  // Test Case 1: Trying to activate a preceding controller when following controller is not activated
+  // --> return error (If STRICT); Preceding controller is still inactive.
+
+  // There is different error and timeout behavior depending on strictness
+  auto const getExpectedBehaviorVals = [&]
+  {
+    if (
+      test_param.strictness ==
+      controller_manager_msgs::srv::SwitchController::Request::STRICT)  // STRICT
+    {
+      return std::make_tuple(
```

Comment:
Can we add this as static variable and not lambda method?
It could be even shorter/easier.

Reply:
Then we should have of course: "expected_strict" and "expected_best_effort" definitions.

Reply:
Where are you reading `test_param`? Can you define this whole structures as a static map? That would be nice and easy to use.

Reply:
Actually static map with struct inside would be the best. This all `std::get<0, 1, 2>` are confusing.